### PR TITLE
synch groups with keycloak

### PR DESCRIPTION
**Pull request by taba90**
- synch keycloak roles not mapped as geostore roles to geostore user groups.

### CODE REVIEW

**Review by offtherailz — COMMENTED** [review]
It could be good to have the option of exluding this group sync or filter via configuration. 
We can open a separate issue for this.
@taba90 let me know if at least exclusion can be introduced in this synchronizing with the team. Otherwise, I will approve.

File: src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/oauth2/google/OAuthGoogleSecurityConfiguration.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ public OAuth2Configuration configuration() {
      */
     @Override
     @Bean(value = "googleOpenIdRestTemplate")
-    @Scope(value = "session", proxyMode = ScopedProxyMode.TARGET_CLASS)
+    @Scope(value = "request", proxyMode = ScopedProxyMode.TARGET_CLASS)
```
Comment (etj) [inline src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/oauth2/google/OAuthGoogleSecurityConfiguration.java:73]: Is this something related to KeyCloak? are we sure it won't break any other oauth2 integration?

File: src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/keycloak/GeoStoreKeycloakAuthProvider.java [hunk 2]
```diff
@@ -103,12 +105,27 @@ public Authentication authenticate(Authentication authentication) throws Authent
...
+
         KeycloakTokenDetails details = new KeycloakTokenDetails(accessTokenStr, refreshToken, expiration);
         details.setIdToken(context.getIdTokenString());
         String username = getUsername(authentication);
-        User user = retrieveUser(username, "", grantedAuthoritiesMapper);
-        if (grantedAuthoritiesMapper != null) user.getGroups().addAll(grantedAuthoritiesMapper.getGroups());
-        if (grantedAuthoritiesMapper != null) user.setRole(grantedAuthoritiesMapper.getRole());
+        Set<UserGroup> keycloakGroups = grantedAuthoritiesMapper != null ? grantedAuthoritiesMapper.getGroups() : new HashSet<>();
+
+        if (configuration.isAutoCreateUser())
```
Comment (etj) [inline src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/keycloak/GeoStoreKeycloakAuthProvider.java:124]: pls add some comment here

**Comment by ale-cristofori** [comment]
@etj connected issue to this PR is https://github.com/geosolutions-it/DevOps/issues/1025

File: src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/oauth2/google/OAuthGoogleSecurityConfiguration.java (hunk 1 above)
Comment (taba90) [reply src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/oauth2/google/OAuthGoogleSecurityConfiguration.java]: the scope value session is an error placed that by me on a previous commit. It might endup creating an HttpSession to store the bean in a stateless context like the geostore rest api. In the context of the Openid filter usage of the above bean it will not change anything regarding the functioning of it, since spring will provide the bean anyway bound to the request instead that bound to the session. I've test locally the google open id authentication after the modification.

File: src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/keycloak/GeoStoreKeycloakAuthProvider.java (hunk 2 above)
Comment (taba90) [reply src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/keycloak/GeoStoreKeycloakAuthProvider.java:125]: fixed.

**Comment by taba90** [comment]
thanks for your review @etj and @offtherailz, feedback applied.

**Comment by offtherailz** [comment]
For Me you can merge, just apply correct labels ( I think we have to backport it too, isn't it?

**Comment by ale-cristofori** [comment]
> For Me you can merge, just apply correct labels, connections etc. .. I think we have to backport it too, isn't it?
as confirmed with @taba90 This has got to be backported to 2022.01.xx, @taba90 should this be PR be backported to 2022.02.xx too?

**Review by etj — APPROVED** [review]
